File: Rock_Paper_Scissors/src/main/java/org/cocharge/rps/RpsGame.java
package org.cocharge.rps;

import org.cocharge.utils.Constants;
import org.cocharge.utils.GameResult;
import org.cocharge.utils.Util;

public class RpsGame {

    public GameResult playGame(){
        String playerA = Constants.ROCK; // Always choose rock as Requested
        String playerB = Util.getRandomChoice();

        return getGameFinalResult (playerA, playerB);
    }

    public GameResult getGameFinalResult(String playerA, String playerB){
        if (playerA.equals(playerB)){
            return  GameResult.DRAW;
        } else if ( playerA.equals(Constants.ROCK) && playerB.equals(Constants.SCISSORS) ||
                    playerA.equals(Constants.PAPER) && playerB.equals(Constants.ROCK) ||
                    playerA.equals(Constants.SCISSORS) && playerB.equals(Constants.PAPER) ){
            return  GameResult.PLAYER_A_WIN;
        } else {
            return GameResult.PLAYER_B_WIN;
        }
    }
}

File: Rock_Paper_Scissors_Kotlin/src/main/kotlin/org/cocharge/rps/RpsGame.kt
package org.cocharge.rps

import org.cocharge.utils.Constants
import org.cocharge.utils.GameResult
import org.cocharge.utils.Util

class RpsGame {

    fun playGame(): GameResult {
        val playerA = Constants.ROCK; // Always Play ROCK
        val playerB = Util.getRandomChoice();

        return getGameFinalResult(playerA, playerB);
    }

    public fun getGameFinalResult(playerA: String,playerB: String): GameResult {
        return when {
            (playerA.equals(playerB)) -> GameResult.DRAW

            (playerA.equals(Constants.ROCK) && playerB.equals(Constants.SCISSORS) ||
             playerA.equals(Constants.PAPER) && playerB.equals(Constants.ROCK) ||
             playerA.equals(Constants.SCISSORS) && playerB.equals(Constants.PAPER)) -> GameResult.PLAYER_A_WIN

            else -> GameResult.PLAYER_B_WIN
        }
    }
}